ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/CIOEngineConfig.kt

```diff
@@ -67,8 +67,13 @@ class EndpointConfig {
      */
     var connectTimeout: Long = 5000
 
+    /**
+     * Socket timeout in millis.
+     */
+    val socketTimeout: Long = 10000
+
     /**
      * Maximum number of connection attempts.
      */
-    var connectRetryAttempts: Int = 5
+    var connectRetryAttempts: Int = 1
```

Comment:
Deprecated

Reply:
We haven't implemented retry on the client level yet, so I think we couldn't deprecate it now.